intranet/apps/eighth/views/profile.py

```diff
@@ -86,7 +88,19 @@ def get_profile_context(request, user_id=None, date=None):
     skipped_ahead = False
 
     blocks_all = EighthBlock.objects.order_by("date", "block_letter").filter(date__gte=date)
-    blocks = blocks_all.filter(date__lt=date_end)
+    #######
+    blocks = []
+    if settings.ENABLE_HYBRID_EIGHTH and not request.user.is_eighth_admin:
+        for b in blocks_all:
```

Comment:
In production, you REALLY want to limit the number of DB queries -- the fact that everything's distributed across 5 servers using Docker means that latency is high. When we first moved Ion to Docker, we had massive latency issues, which took some tuning (#959, #960, #964) to work out. (The "SQL" tab of the Django Debug Toolbar is *extremely* useful.)

It's been a while since I wrote Django queries, but I suspect this whole section can be simplified down to 1 or 2 big queries. Something like this (untested; more an outline than a working query): `blocks = blocks_all.filter(eighthscheduledactivity_set__in=EighthScheduledActivity.objects.filter(activity__name="z - Hybrid Sticky").exclude(members_username=request.user.id)).distinct().order_by("date", "block_letter")`